consensus/istanbul/core/core.go

```diff
@@ -173,7 +173,14 @@ func (c *core) commit() {
 		committedSeals := make([][]byte, c.current.Commits.Size())
 		for i, v := range c.current.Commits.Values() {
 			committedSeals[i] = make([]byte, types.IstanbulExtraCommittedSeal)
-			copy(committedSeals[i][:], v.CommittedSeal[:])
+
+			var commit *istanbul.CommittedSubject
+			err := v.Decode(&commit)
+			if err != nil {
+				panic(fmt.Sprintf("commit: error in decoding committed subject for address %s", hex.EncodeToString(v.Address[:])))
```

Comment:
As we discussed yesterday, would be better to return an error. Fine to save this for a follow-up PR as I believe we have an issue around this